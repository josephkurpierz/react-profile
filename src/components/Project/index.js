import React from "react";
import parkour from "../../assets/projects/Parkour-landing-page.png";
import pizza from "../../assets/projects/pizza-hunt-landing-page.png";
import photoPort from "../../assets/projects/photo-port-landing-page.png";
import zooKeepr from "../../assets/projects/zookeepr-landing-page.png";
import dogs from "../../assets/projects/dogs-dogs-dogs-landing-page.png";

function Project() {
  const projects = [
    {
      name: "Parkour",
      thumbnail: parkour,
      repo: "https://github.com/josephkurpierz/parkour",
      deployed: "https://parkour-review.herokuapp.com/",
    },
    {
      name: "Pizza Hunt",
      thumbnail: pizza,
      repo: "https://github.com/josephkurpierz/pizza-hunt",
      deployed: "https://nameless-shore-70920.herokuapp.com/",
    },
    {
      name: "Photo Portfolio",
      thumbnail: photoPort,
      repo: "https://github.com/josephkurpierz/photo-port",
      deployed: "https://josephkurpierz.github.io/photo-port/",
    },
    {
      name: "Zoo Keepr",
      thumbnail: zooKeepr,
      repo: "https://github.com/josephkurpierz/zookeepr",
      deployed: "https://fast-sea-99289.herokuapp.com/",
    },
    {
      name: "Dogs-Dogs-Dogs",
      thumbnail: dogs,
      repo: "https://github.com/josephkurpierz/dogs-dogs-dogs",
      deployed: "https://taylorstrubhar.github.io/dogs-dogs-dogs/",
    },
  ];
  return (
    <section>
      <h2 className="section-title">My Projects</h2>
      <div className="project-list">
        {projects.map((project) => (
          <div className="project-card" key={project.name}>
              <a className='links' href={project.repo}>{project.name}</a><br/>
              <a className='links'href={project.deployed}>Live deployment</a>
            <div
              className="project-img"
              style={{ backgroundImage: `url(${project.thumbnail})` }}
            >
            </div>
          </div>
        ))}
      </div>
    </section>
  );
}

export default Project;
