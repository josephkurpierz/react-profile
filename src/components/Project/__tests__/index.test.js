import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Project from '..'
import parkour from '../../../assets/projects/parkour-landing-page.png';

const project = {
  name: "Parkour",
  thumbnail: parkour,
  repo: "https://github.com/josephkurpierz/parkour",
  deployed: "https://parkour-review.herokuapp.com/",
};

afterEach(cleanup);

describe('Project is rendering',()=>{
  it('renders', () => {
    render(<Project />);
  });
});

it('matches snapshot', () => {
  const { asFragment } = render(<Project/>)
  expect(asFragment()).toMatchSnapshot();
});

it('renders', ()=>{
  render(<Project/>)
  expect(screen.getByTestId('h1tag')).toHaveTextContent('Portraits');
})