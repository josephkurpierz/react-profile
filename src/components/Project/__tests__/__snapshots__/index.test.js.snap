// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2>
      My Projects
    </h2>
    <div>
      <div>
        <div
          class="projectCard"
          style="background-image: url(Parkour-landing-page.png);"
        >
          <a
            href="https://github.com/josephkurpierz/parkour"
          >
            Parkour
          </a>
          <a
            href="https://parkour-review.herokuapp.com/"
          >
            Live deployment
          </a>
        </div>
      </div>
      <div>
        <div
          class="projectCard"
          style="background-image: url(pizza-hunt-landing-page.png);"
        >
          <a
            href="https://github.com/josephkurpierz/pizza-hunt"
          >
            Pizza Hunt
          </a>
          <a
            href="https://nameless-shore-70920.herokuapp.com/"
          >
            Live deployment
          </a>
        </div>
      </div>
      <div>
        <div
          class="projectCard"
          style="background-image: url(photo-port-landing-page.png);"
        >
          <a
            href="https://github.com/josephkurpierz/photo-port"
          >
            Photo Portfolio
          </a>
          <a
            href="https://josephkurpierz.github.io/photo-port/"
          >
            Live deployment
          </a>
        </div>
      </div>
      <div>
        <div
          class="projectCard"
          style="background-image: url(zookeepr-landing-page.png);"
        >
          <a
            href="https://github.com/josephkurpierz/zookeepr"
          >
            Zoo Keepr
          </a>
          <a
            href="https://fast-sea-99289.herokuapp.com/"
          >
            Live deployment
          </a>
        </div>
      </div>
      <div>
        <div
          class="projectCard"
          style="background-image: url(dogs-dogs-dogs-landing-page.png);"
        >
          <a
            href="https://github.com/josephkurpierz/dogs-dogs-dogs"
          >
            Dogs-Dogs-Dogs
          </a>
          <a
            href="https://taylorstrubhar.github.io/dogs-dogs-dogs/"
          >
            Live deployment
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
